pandas/core/groupby/groupby.py

```diff
@@ -2133,22 +2153,37 @@ def var(
 
             return self._numba_agg_general(sliding_var, engine_kwargs, ddof)
         else:
+            ignore_failures = numeric_only is lib.no_default
+            numeric_only = self._resolve_numeric_only(numeric_only)
             if ddof == 1:
-                numeric_only = self._resolve_numeric_only(lib.no_default)
                 return self._cython_agg_general(
                     "var",
                     alt=lambda x: Series(x).var(ddof=ddof),
                     numeric_only=numeric_only,
+                    ignore_failures=ignore_failures,
                 )
             else:
                 func = lambda x: x.var(ddof=ddof)
-                with self._group_selection_context():
-                    return self._python_agg_general(func)
+                if numeric_only:
+                    nonnumeric_exclusions = frozenset(
+                        self.obj.columns.difference(self.exclusions).difference(
+                            self.obj._get_numeric_data().columns
+                        )
+                    )
+                else:
+                    nonnumeric_exclusions = frozenset()
+                with com.temp_setattr(
+                    self, "exclusions", self.exclusions | nonnumeric_exclusions
```

Comment:
what is the purpose here?

Reply:
Thanks - I was able to remove this hack by setting the appropriate value of raise_on_typeerror.